app/lib/iiif_manifest_factory.rb

```diff
@@ -1,22 +1,65 @@
 # frozen_string_literal: true
 
 class IiifManifestFactory
-  attr_reader :oid, :manifest, :parent_object
+  attr_reader :oid, :manifest, :parent_object, :mets_doc
 
   require 'iiif/presentation'
 
   def initialize(oid)
     @oid = oid
     @parent_object = ParentObject.find_by(oid: oid)
+    goobi_object = GoobiXmlImport.find_by(oid: oid)
+    iiif_info = { oid: oid, reason: "No METs xml available for #{oid}" }
+    return Rails.logger.error("Unable to create iiif manifest for: #{iiif_info.to_json}") unless goobi_object
```

Comment:
@maxkadel I think this needs more Micheal Bay: 💥  I feel like a ActiveRecordNotFound exception is probably appropriate here.